src/services/badgeModels/badgeModels.ts

```diff
@@ -33,15 +40,40 @@ export class BadgeModelsService extends TheBadgeSDKConfig implements BadgeModels
    * @param badgeModelId
    */
   async getById(badgeModelId: string): Promise<BadgeModelByIdQuery> {
-    return await this.subgraph.badgeModelById({ badgeModelId })
+    return await this.subgraph.badgeModelById({ id: badgeModelId })
   }
 
   /**
    * Obtain the metadata of a badge model giving its id
    *
    * @param badgeModelId
    */
-  async getMetadataOfBadgeModel(badgeModelId: string): Promise<BadgeModelByIdQuery> {
+  async getMetadataOfBadgeModel(badgeModelId: string): Promise<BadgeModelMetadataByIdQuery> {
     return await this.subgraph.badgeModelMetadataById({ id: badgeModelId })
   }
+
+  /**
+   * Get evidence requirements of a badge model giving its id
+   *
+   * @param badgeModelId
+   * @returns an Array<MetadataColumn> or null if not found
+   */
+  public async getEvidenceRequirementsOfBadgeModel(badgeModelId: string): Promise<Array<MetadataColumn> | null> {
+    // take ipfs uri from metadata of the badge model
+    const badgeModelMetadataResponse = await this.getMetadataOfBadgeModel(badgeModelId)
+    const ipfsDataUri = badgeModelMetadataResponse?.badgeModelKlerosMetaData?.registrationUri
+    if (!ipfsDataUri) {
+      return null
```

Comment:
I think that we should do `throw new Error("Missing RegistrationUri for the given model ID, provide a valid ModelID")`, return just `null` can be confusing. The same for the other cases where you are doing a check

Reply:
fixed